frontend/src/components/CreateForm.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { useListingsContext } from '../hooks/useListingsContext';
+
+const CreateForm = (props) => {
+    const { dispatch } = useListingsContext();
+    const id = localStorage.getItem('id');
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        const { title, organisationName, description, requirement, firstLine, city, postcode, neededByDate } = e.target.elements;
+        const listing = {
+            title: title.value,
+            organisationName: organisationName.value,
+            description: description.value,
+            requirement: requirement.value,
+            address: {
+                firstLine: firstLine.value,
+                city: city.value,
+                postcode: postcode.value
+            },
+            neededByDate: neededByDate.value
+        }
+        // props is passed in from ListingsFeed.js
+        props.createListing(listing);
+    }
+
+    return (
+        <form onSubmit={handleSubmit}>
+            <label htmlFor="title">Title</label>
+            <input type="text" name="title" id="title" />
+            <label htmlFor="organisationName">Organisation Name</label>
+            <input type="text" name="organisationName" id="organisationName" />
+            <label htmlFor="description">Description</label>
+            <input type="text" name="description" id="description" />
+            <label htmlFor="requirement">Requirement</label>
+            <input type="text" name="requirement" id="requirement" />
```

Comment:
Could we have 'requirement' input as a dropdown list please?

- Volunteering
- Donation of Goods

Reply:
yeah was going to say the same